multicluster/controllers/multicluster/member/stale_controller.go

```diff
@@ -375,80 +366,155 @@ func (c *StaleResCleanupController) cleanupClusterInfoResourceExport(ctx context
 	return nil
 }
 
-func (c *StaleResCleanupController) cleanupMemberClusterAnnounces(ctx context.Context) error {
-	memberClusterAnnounceList := &mcsv1alpha1.MemberClusterAnnounceList{}
-	if err := c.List(ctx, memberClusterAnnounceList, &client.ListOptions{}); err != nil {
+// Run starts the StaleResCleanupController and blocks until stopCh is closed.
+func (c *StaleResCleanupController) Run(stopCh <-chan struct{}) {
+	klog.InfoS("Starting StaleResCleanupController")
+	defer klog.InfoS("Shutting down StaleResCleanupController")
+
+	ctx, _ := wait.ContextForChannel(stopCh)
+	retry.OnError(common.CleanUpRetry, func(err error) bool { return true },
+		func() error {
+			return c.CleanUp(ctx)
+		})
+
+	go func() {
```

Comment:
Do we need a separate goroutinue here?

Reply:
Removed.

Reply:
We still have the goroutine. My question is if we do cleanup (at line 389) first, do we still need a goroutinue to wait for `commonAreaCreationCh`? Could we remove the goroutine and `<-stopCh` at the end?